src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/SearchController.kt

```diff
@@ -111,12 +111,6 @@ class SearchController(
   }
 
   private fun buildIdentifierRecord(personEntity: PersonEntity): PersonIdentifierRecord? {
-    return when (personEntity.sourceSystem) {
-      SourceSystemType.DELIUS -> PersonIdentifierRecord(id = personEntity.crn!!, SourceSystemType.DELIUS.name)
-      SourceSystemType.NOMIS -> PersonIdentifierRecord(id = personEntity.prisonNumber!!, SourceSystemType.NOMIS.name)
-      SourceSystemType.COMMON_PLATFORM -> PersonIdentifierRecord(id = personEntity.defendantId!!, SourceSystemType.COMMON_PLATFORM.name)
-      SourceSystemType.LIBRA -> PersonIdentifierRecord(id = personEntity.defendantId!!, SourceSystemType.LIBRA.name)
-      else -> null
-    }
+    return personEntity.extractSourceSystemId()?.let { PersonIdentifierRecord(id = it, personEntity.sourceSystem.name) }
```

Comment:
Could this return a pair or a small object containing the source system ID and the identifier? As far as I can tell from a quick glance we always use both after calling this method

Reply:
I think it is some what redundant as we would be extracting that from the record which already has the correct `source system` so no extra logic to figure it out. Just need to know what identifier to return.